include/ensmallen_bits/callbacks/report.hpp

```diff
@@ -149,6 +149,15 @@ class Report
     output << coordinates.n_cols << std::endl;
     output << std::endl;
 
+    // If we did not take any steps, at least fill what the initial objective
+    // was.
+    const bool tookStep = (objectives.size() > 0);
+    if (objectives.size() == 0)
+    {
+      objectives.push_back(function.Evaluate(coordinates));
```

Comment:
This will fail since not every function implements `Evaluate(coordinates)`, I think in most cases an optimizer will run Evaluate before the actual optimization starts, so one solution might be to check if `evaluateCalls > 0` and then return `objective` which is set during a call of `Evaluate`. In case it's not set we could return `NaN`.

Reply:
Ohh, good point.  I think that's a better idea.  Updated in 50439fb; let me know what you think. :+1: